### PR TITLE
refactor bindX & fix bindV

### PR DESCRIPTION
**Pull request by vicb**
Easier to understand.

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

File: packages/core/src/render3/instructions.ts [hunk 1]
```diff
@@ -1446,22 +1454,12 @@ export function bind1(prefix: string, value: any, suffix: string): string|NO_CHA
   return bind(value) === NO_CHANGE ? NO_CHANGE : prefix + stringify(value) + suffix;
 }
 
-/**
- * Create an interpolation bindings with 2 arguments.
- *
- * @param prefix
- * @param v0 value checked for change
- * @param i0
```
Comment (mhevery) [inline packages/core/src/render3/instructions.ts]: Why did you remove the doc?

**Review by mhevery — APPROVED** [review]

File: packages/core/src/render3/instructions.ts (hunk 1 above)
Comment (vicb) [reply packages/core/src/render3/instructions.ts]: Because I don't see any added value in 
```
- * @param prefix
 - * @param v0
 - * @param i0
 - * @param v1
 - * @param i1
 - * @param v2
 - * @param i2
 - * @param v3
 - * @param i3
 - * @param v4
 - * @param suffix
```

File: packages/core/src/render3/instructions.ts [hunk 2]
```diff
@@ -1372,33 +1372,38 @@ export const NO_CHANGE = {} as NO_CHANGE;
  * If any of the arguments change, then the interpolation is concatenated
  * and causes an update.
  *
- * @param values an array of values to diff.
+ * `values`:
+ * - has static text at even indexes,
+ * - has evaluated expressions at odd indexes (could be NO_CHANGE).
  */
 export function bindV(values: any[]): string|NO_CHANGE {
+  ngDevMode && assertLessThan(2, values.length, 'should have at least 3 values');
```
Comment (chuckjaz) [inline packages/core/src/render3/instructions.ts]: Should we assert that it has at least 9 values as the more optimal binds should be called instead?

**Review by chuckjaz — APPROVED** [review]

File: packages/core/src/render3/instructions.ts (hunk 2 above)
Comment (vicb) [reply packages/core/src/render3/instructions.ts]: The code works with 3+ values but I agree it should not be called with less than 9.
We should probably have a mechanism to check that (out of scope of this PR)
